packages/@aws-cdk/aws-scheduler-targets-alpha/lib/lambda-invoke.ts

```diff
@@ -1,34 +1,23 @@
 import { ISchedule, IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
-import { Names } from 'aws-cdk-lib';
 import { IRole } from 'aws-cdk-lib/aws-iam';
 import * as lambda from 'aws-cdk-lib/aws-lambda';
 import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
-import { sameEnvDimension } from './util';
 
 /**
  * Use an AWS Lambda function as a target for AWS EventBridge Scheduler.
  */
 export class LambdaInvoke extends ScheduleTargetBase implements IScheduleTarget {
+  private readonly func: lambda.IFunction;
+
   constructor(
-    private readonly func: lambda.IFunction,
-    private readonly props: ScheduleTargetBaseProps,
+    func: lambda.IFunction,
+    props: ScheduleTargetBaseProps,
   ) {
     super(props, func.functionArn);
+    this.func = func;
   }
 
-  protected addTargetActionToRole(schedule: ISchedule, role: IRole): void {
-    if (!sameEnvDimension(this.func.env.region, schedule.env.region)) {
```

Comment:
While this would fix the problem mentioned in the issue, I don't think this is a good fix in general since it removes check even for functions and scheduler with known account & regions.

Reply:
That is correct. The reasoning to remove the check is explained [here](https://github.com/aws/aws-cdk/pull/29615).